src/software/new_geom/util/intersects.h

```diff
@@ -14,6 +14,8 @@
 
 /**
  * Returns true if the segment intersects the polygon, false otherwise.
+ * A segment is considered to intersect the polygon iff it has one point
```

Comment:
This seems to describe the old behaviour of the function, not the fixed one

Reply:
clarified the comment

Reply:
If I'm understanding this fix correctly, is it no longer the case that a point on the segment needs to be inside the polygon to return `true`?

Reply:
no, the segment needs to have a point that inside the polygon and a different point that outside the polygon

Reply:
But wouldn't that mean a segment that passes through the "corner" of a polygon (without either of its endpoints being inside the polygon) does not count as intersecting? This wouldn't match the implementation to my understanding.

Reply:
![Screenshot from 2020-04-28 13-52-55](https://user-images.githubusercontent.com/21165407/80536663-93059c80-8957-11ea-9a3a-e535fd063dc2.png)
I'm referring to something like this

Reply:
I supposed I'm struggling to explain this because of the difference between endpoints and points that are part of the segment. I'm interested in the all the points that are part of the segment, i.e. _not_ just the end points. 

Here's a more rigorous definition:
Define:
* _S_ to be the set of points that belong to the `segment`, including the endpoints and all the points between the endpoints.
* `polygon` the Polygon

Intersects(`polygon`, `segment`) is true iff (∃ points _a_, _b_ in _S_, s.t. `polygon`.contains(_a_) is true and  `polygon`.contains(_b_) is false

Reply:
I think I understand now. However, I think a segment being entirely contained in a polygon should return `true` based on the definition of a intersects b is `a ∩ b ≠ ∅ ` (https://en.wikipedia.org/wiki/Spatial_relation#Topological_relations).

Is there a specific use case that requires the above case to return `false`?

Reply:
I like that reference a lot. I changed the behaviour to match that. I've also updated contains to match that reference -> contains means one object is completely inside of the other.